r/R/query-engine.R

```diff
@@ -59,6 +59,10 @@ ExecPlan <- R6Class("ExecPlan",
     Scan = function(dataset) {
       # Handle arrow_dplyr_query
       if (inherits(dataset, "arrow_dplyr_query")) {
+        if(inherits(dataset$.data, "RecordBatchReader")) {
+          return(ExecNode_ReadFromRecordBatchReader(self, dataset$.data))
+        }
```

Comment:
This doesn't (yet) work, I'm getting a segfault

```
Warning: stack imbalance in '[[', 215 then 216
Warning: stack imbalance in 'is.null', 213 then 220
Warning: stack imbalance in '!', 212 then 219
Warning: stack imbalance in '&&', 210 then 217
Error: C stack usage  17587191479824 is too close to the limit
Warning: stack imbalance in '&&', 208 then 215
Warning: stack imbalance in '(', 207 then 214
Warning: stack imbalance in '||', 205 then 212
Warning: stack imbalance in 'if', 203 then 210
Warning: stack imbalance in '{', 199 then 218
Warning: stack imbalance in '{', 192 then 211
Warning: stack imbalance in 'is.null', 187 then 206
Warning: stack imbalance in '!', 186 then 205
Warning: stack imbalance in '&&', 184 then 203
Warning: stack imbalance in '&&', 182 then 201
Warning: stack imbalance in '(', 181 then 200
Warning: stack imbalance in '||', 179 then 198
Warning: stack imbalance in 'if', 177 then 196
Warning: stack imbalance in '{', 173 then 192
Warning: stack imbalance in '<-', 168 then 187
Warning: stack imbalance in '{', 164 then 183
Warning: stack imbalance in 'if', 162 then 181

 *** caught segfault ***
address 0x0, cause 'memory not mapped'
Warning: stack imbalance in '>', 183 then 187
```

I'm digging into this now